File: domain/src/main/java/com/aceplus/domain/model/forApi/company/CompanyInformation.java
package com.aceplus.domain.model.forApi.company;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by phonelin on 4/6/17.
 */

public class CompanyInformation {

    @SerializedName("Id")
    @Expose
    private Integer id;
    @SerializedName("Description")
    @Expose
    private String description;
    @SerializedName("MainDBName")
    @Expose
    private String mainDBName;
    @SerializedName("HomeCurrencyId")
    @Expose
    private Integer homeCurrencyId;
    @SerializedName("MultiCurrency")
    @Expose
    private Integer multiCurrency;
    @SerializedName("StartDate")
    @Expose
    private String startDate;
    @SerializedName("EndDate")
    @Expose
    private String endDate;
    @SerializedName("AutoGenerate")
    @Expose
    private Integer autoGenerate;
    @SerializedName("CompanyName")
    @Expose
    private String companyName;
    @SerializedName("ShortName")
    @Expose
    private String shortName;
    @SerializedName("ContactPerson")
    @Expose
    private String contactPerson;
    @SerializedName("Address")
    @Expose
    private String address;
    @SerializedName("Email")
    @Expose
    private String email;
    @SerializedName("Website")
    @Expose
    private String website;
    @SerializedName("SerialNumber")
    @Expose
    private String serialNumber;
    @SerializedName("PhoneNumber")
    @Expose
    private String phoneNumber;
    @SerializedName("IsSeparator")
    @Expose
    private Integer isSeparator;
    @SerializedName("Amount_Format")
    @Expose
    private Integer amountFormat;
    @SerializedName("Price_Format")
    @Expose
    private Integer priceFormat;
    @SerializedName("Quantity_Format")
    @Expose
    private Integer quantityFormat;
    @SerializedName("Rate_Format")
    @Expose
    private Integer rateFormat;
    @SerializedName("ValuationMethod")
    @Expose
    private String valuationMethod;
    @SerializedName("Font")
    @Expose
    private String font;
    @SerializedName("ReportFont")
    @Expose
    private String reportFont;
    @SerializedName("ReceiptVoucher")
    @Expose
    private String receiptVoucher;
    @SerializedName("prnPort")
    @Expose
    private String prnPort;
    @SerializedName("POSVoucherFooter1")
    @Expose
    private String pOSVoucherFooter1;
    @SerializedName("POSVoucherFooter2")
    @Expose
    private String pOSVoucherFooter2;
    @SerializedName("IsStockAutoGenerate")
    @Expose
    private Integer isStockAutoGenerate;
    @SerializedName("PCCount")
    @Expose
    private String pCCount;
    @SerializedName("ExpiredMonth")
    @Expose
    private String expiredMonth;
    @SerializedName("Paidstatus")
    @Expose
    private String paidstatus;
    @SerializedName("H1")
    @Expose
    private String h1;
    @SerializedName("H2")
    @Expose
    private String h2;
    @SerializedName("H3")
    @Expose
    private String h3;
    @SerializedName("H4")
    @Expose
    private String h4;
    @SerializedName("F1")
    @Expose
    private String f1;
    @SerializedName("F2")
    @Expose
    private String f2;
    @SerializedName("F3")
    @Expose
    private String f3;
    @SerializedName("F4")
    @Expose
    private String f4;
    @SerializedName("Tax")
    @Expose
    private Integer tax;
    @SerializedName("Branch_Code")
    @Expose
    private String branchCode;
    @SerializedName("Branch_Name")
    @Expose
    private String branchName;
    @SerializedName("HBCode")
    @Expose
    private String hBCode;
    @SerializedName("Active")
    @Expose
    private Integer active;
    @SerializedName("TS")
    @Expose
    private String tS;
    @SerializedName("CreatedDate")
    @Expose
    private String createdDate;
    @SerializedName("CreatedUserID")
    @Expose
    private Integer createdUserID;
    @SerializedName("UpdatedDate")
    @Expose
    private String updatedDate;
    @SerializedName("UpdatedUserID")
    @Expose
    private String updatedUserID;
    @SerializedName("CreditSale")
    @Expose
    private Integer creditSale;
    @SerializedName("UseCombo")
    @Expose
    private Integer useCombo;
    @SerializedName("LastDayCloseDate")
    @Expose
    private String lastDayCloseDate;
    @SerializedName("LastUpdateInvoiceDate")
    @Expose
    private String lastUpdateInvoiceDate;
    @SerializedName("CompanyType")
    @Expose
    private String companyType;
    @SerializedName("CompanyCode")
    @Expose
    private String companyCode;
    @SerializedName("StartTime")
    @Expose
    private String startTime;
    @SerializedName("EndTime")
    @Expose
    private String endTime;
    @SerializedName("BranchId")
    @Expose
    private Integer branchId;
    @SerializedName("PrintCopy")
    @Expose
    private Integer printCopy;
    @SerializedName("TaxType")
    @Expose
    private String taxType;
    @SerializedName("BalanceControl")
    @Expose
    private String balanceControl;
    @SerializedName("TransactionAutoGenerate")
    @Expose
    private Integer transactionAutoGenerate;

    @SerializedName("CommTexRegNo")
    @Expose
    private String commonTaxRegNo;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getMainDBName() {
        return mainDBName;
    }

    public void setMainDBName(String mainDBName) {
        this.mainDBName = mainDBName;
    }

    public Integer getHomeCurrencyId() {
        return homeCurrencyId;
    }

    public void setHomeCurrencyId(Integer homeCurrencyId) {
        this.homeCurrencyId = homeCurrencyId;
    }

    public Integer getMultiCurrency() {
        return multiCurrency;
    }

    public void setMultiCurrency(Integer multiCurrency) {
        this.multiCurrency = multiCurrency;
    }

    public String getStartDate() {
        return startDate;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public String getEndDate() {
        return endDate;
    }

    public void setEndDate(String endDate) {
        this.endDate = endDate;
    }

    public Integer getAutoGenerate() {
        return autoGenerate;
    }

    public void setAutoGenerate(Integer autoGenerate) {
        this.autoGenerate = autoGenerate;
    }

    public String getCompanyName() {
        return companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public String getShortName() {
        return shortName;
    }

    public void setShortName(String shortName) {
        this.shortName = shortName;
    }

    public String getContactPerson() {
        return contactPerson;
    }

    public void setContactPerson(String contactPerson) {
        this.contactPerson = contactPerson;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public void setSerialNumber(String serialNumber) {
        this.serialNumber = serialNumber;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public Integer getIsSeparator() {
        return isSeparator;
    }

    public void setIsSeparator(Integer isSeparator) {
        this.isSeparator = isSeparator;
    }

    public Integer getAmountFormat() {
        return amountFormat;
    }

    public void setAmountFormat(Integer amountFormat) {
        this.amountFormat = amountFormat;
    }

    public Integer getPriceFormat() {
        return priceFormat;
    }

    public void setPriceFormat(Integer priceFormat) {
        this.priceFormat = priceFormat;
    }

    public Integer getQuantityFormat() {
        return quantityFormat;
    }

    public void setQuantityFormat(Integer quantityFormat) {
        this.quantityFormat = quantityFormat;
    }

    public Integer getRateFormat() {
        return rateFormat;
    }

    public void setRateFormat(Integer rateFormat) {
        this.rateFormat = rateFormat;
    }

    public String getValuationMethod() {
        return valuationMethod;
    }

    public void setValuationMethod(String valuationMethod) {
        this.valuationMethod = valuationMethod;
    }

    public String getFont() {
        return font;
    }

    public void setFont(String font) {
        this.font = font;
    }

    public String getReportFont() {
        return reportFont;
    }

    public void setReportFont(String reportFont) {
        this.reportFont = reportFont;
    }

    public String getReceiptVoucher() {
        return receiptVoucher;
    }

    public void setReceiptVoucher(String receiptVoucher) {
        this.receiptVoucher = receiptVoucher;
    }

    public String getPrnPort() {
        return prnPort;
    }

    public void setPrnPort(String prnPort) {
        this.prnPort = prnPort;
    }

    public String getPOSVoucherFooter1() {
        return pOSVoucherFooter1;
    }

    public void setPOSVoucherFooter1(String pOSVoucherFooter1) {
        this.pOSVoucherFooter1 = pOSVoucherFooter1;
    }

    public String getPOSVoucherFooter2() {
        return pOSVoucherFooter2;
    }

    public void setPOSVoucherFooter2(String pOSVoucherFooter2) {
        this.pOSVoucherFooter2 = pOSVoucherFooter2;
    }

    public Integer getIsStockAutoGenerate() {
        return isStockAutoGenerate;
    }

    public void setIsStockAutoGenerate(Integer isStockAutoGenerate) {
        this.isStockAutoGenerate = isStockAutoGenerate;
    }

    public String getPCCount() {
        return pCCount;
    }

    public void setPCCount(String pCCount) {
        this.pCCount = pCCount;
    }

    public String getExpiredMonth() {
        return expiredMonth;
    }

    public void setExpiredMonth(String expiredMonth) {
        this.expiredMonth = expiredMonth;
    }

    public String getPaidstatus() {
        return paidstatus;
    }

    public void setPaidstatus(String paidstatus) {
        this.paidstatus = paidstatus;
    }

    public String getH1() {
        return h1;
    }

    public void setH1(String h1) {
        this.h1 = h1;
    }

    public String getH2() {
        return h2;
    }

    public void setH2(String h2) {
        this.h2 = h2;
    }

    public String getH3() {
        return h3;
    }

    public void setH3(String h3) {
        this.h3 = h3;
    }

    public String getH4() {
        return h4;
    }

    public void setH4(String h4) {
        this.h4 = h4;
    }

    public String getF1() {
        return f1;
    }

    public void setF1(String f1) {
        this.f1 = f1;
    }

    public String getF2() {
        return f2;
    }

    public void setF2(String f2) {
        this.f2 = f2;
    }

    public String getF3() {
        return f3;
    }

    public void setF3(String f3) {
        this.f3 = f3;
    }

    public String getF4() {
        return f4;
    }

    public void setF4(String f4) {
        this.f4 = f4;
    }

    public Integer getTax() {
        return tax;
    }

    public void setTax(Integer tax) {
        this.tax = tax;
    }

    public String getBranchCode() {
        return branchCode;
    }

    public void setBranchCode(String branchCode) {
        this.branchCode = branchCode;
    }

    public String getBranchName() {
        return branchName;
    }

    public void setBranchName(String branchName) {
        this.branchName = branchName;
    }

    public String getHBCode() {
        return hBCode;
    }

    public void setHBCode(String hBCode) {
        this.hBCode = hBCode;
    }

    public Integer getActive() {
        return active;
    }

    public void setActive(Integer active) {
        this.active = active;
    }

    public String getTS() {
        return tS;
    }

    public void setTS(String tS) {
        this.tS = tS;
    }

    public String getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(String createdDate) {
        this.createdDate = createdDate;
    }

    public Integer getCreatedUserID() {
        return createdUserID;
    }

    public void setCreatedUserID(Integer createdUserID) {
        this.createdUserID = createdUserID;
    }

    public String getUpdatedDate() {
        return updatedDate;
    }

    public void setUpdatedDate(String updatedDate) {
        this.updatedDate = updatedDate;
    }

    public String getUpdatedUserID() {
        return updatedUserID;
    }

    public void setUpdatedUserID(String updatedUserID) {
        this.updatedUserID = updatedUserID;
    }

    public Integer getCreditSale() {
        return creditSale;
    }

    public void setCreditSale(Integer creditSale) {
        this.creditSale = creditSale;
    }

    public Integer getUseCombo() {
        return useCombo;
    }

    public void setUseCombo(Integer useCombo) {
        this.useCombo = useCombo;
    }

    public String getLastDayCloseDate() {
        return lastDayCloseDate;
    }

    public void setLastDayCloseDate(String lastDayCloseDate) {
        this.lastDayCloseDate = lastDayCloseDate;
    }

    public String getLastUpdateInvoiceDate() {
        return lastUpdateInvoiceDate;
    }

    public void setLastUpdateInvoiceDate(String lastUpdateInvoiceDate) {
        this.lastUpdateInvoiceDate = lastUpdateInvoiceDate;
    }

    public String getCompanyType() {
        return companyType;
    }

    public void setCompanyType(String companyType) {
        this.companyType = companyType;
    }

    public String getCompanyCode() {
        return companyCode;
    }

    public void setCompanyCode(String companyCode) {
        this.companyCode = companyCode;
    }

    public String getStartTime() {
        return startTime;
    }

    public void setStartTime(String startTime) {
        this.startTime = startTime;
    }

    public String getEndTime() {
        return endTime;
    }

    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }

    public Integer getBranchId() {
        return branchId;
    }

    public void setBranchId(Integer branchId) {
        this.branchId = branchId;
    }

    public Integer getPrintCopy() {
        return printCopy;
    }

    public void setPrintCopy(Integer printCopy) {
        this.printCopy = printCopy;
    }

    public String getTaxType() {
        return taxType;
    }

    public void setTaxType(String taxType) {
        this.taxType = taxType;
    }

    public String getBalanceControl() {
        return balanceControl;
    }

    public void setBalanceControl(String balanceControl) {
        this.balanceControl = balanceControl;
    }

    public Integer getTransactionAutoGenerate() {
        return transactionAutoGenerate;
    }

    public void setTransactionAutoGenerate(Integer transactionAutoGenerate) {
        this.transactionAutoGenerate = transactionAutoGenerate;
    }

    public String  getCommonTaxRegNo() {
        return commonTaxRegNo;
    }

    public void setCommonTaxRegNo(String commonTaxRegNo) {
        this.commonTaxRegNo = commonTaxRegNo;
    }
}

File: domain/src/main/java/com/aceplus/domain/entity/CompanyInformation.kt
package com.aceplus.domain.entity

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "company_information")
class CompanyInformation {

    @PrimaryKey
    @ColumnInfo(name = "id")
    @SerializedName("id")
    @Expose
    var id: Int = 0

    @ColumnInfo(name = "description")
    @SerializedName("description")
    @Expose
    var description: String? = null

    @ColumnInfo(name = "main_db_name")
    @SerializedName("main_db_name")
    @Expose
    var main_db_name: String? = null

    @ColumnInfo(name = "home_currency_id")
    @SerializedName("home_currency_id")
    @Expose
    var home_currency_id: String? = null

    @ColumnInfo(name = "multi_currency")
    @SerializedName("multi_currency")
    @Expose
    var multi_currency: String? = null

    @ColumnInfo(name = "start_date")
    @SerializedName("start_date")
    @Expose
    var start_date: String? = null

    @ColumnInfo(name = "end_date")
    @SerializedName("end_date")
    @Expose
    var end_date: String? = null

    @ColumnInfo(name = "auto_generate")
    @SerializedName("auto_generate")
    @Expose
    var auto_generate: String? = null

    @ColumnInfo(name = "company_name")
    @SerializedName("company_name")
    @Expose
    var company_name: String? = null

    @ColumnInfo(name = "short_name")
    @SerializedName("short_name")
    @Expose
    var short_name: String? = null

    @ColumnInfo(name = "contact_person")
    @SerializedName("contact_person")
    @Expose
    var contact_person: String? = null

    @ColumnInfo(name = "address")
    @SerializedName("address")
    @Expose
    var address: String? = null

    @ColumnInfo(name = "email")
    @SerializedName("email")
    @Expose
    var email: String? = null

    @ColumnInfo(name = "website")
    @SerializedName("website")
    @Expose
    var website: String? = null

    @ColumnInfo(name = "serial_number")
    @SerializedName("serial_number")
    @Expose
    var serial_number: String? = null

    @ColumnInfo(name = "phone_number")
    @SerializedName("phone_number")
    @Expose
    var phone_number: String? = null

    @ColumnInfo(name = "is_separator")
    @SerializedName("is_separator")
    @Expose
    var separator: String? = null

    @ColumnInfo(name = "amount_format")
    @SerializedName("amount_format")
    @Expose
    var amount_format: String? = null

    @ColumnInfo(name = "price_format")
    @SerializedName("price_format")
    @Expose
    var price_format: String? = null

    @ColumnInfo(name = "quantity_format")
    @SerializedName("quantity_format")
    @Expose
    var quantity_format: String? = null

    @ColumnInfo(name = "rate_format")
    @SerializedName("rate_format")
    @Expose
    var rate_format: String? = null

    @ColumnInfo(name = "valuation_method")
    @SerializedName("valuation_method")
    @Expose
    var valuation_method: String? = null

    @ColumnInfo(name = "font")
    @SerializedName("font")
    @Expose
    var font: String? = null

    @ColumnInfo(name = "report_font")
    @SerializedName("report_font")
    @Expose
    var report_font: String? = null

    @ColumnInfo(name = "receipt_voucher")
    @SerializedName("receipt_voucher")
    @Expose
    var receipt_voucher: String? = null

    @ColumnInfo(name = "print_port")
    @SerializedName("print_port")
    @Expose
    var print_port: String? = null

    @ColumnInfo(name = "pos_voucher_footer1")
    @SerializedName("pos_voucher_footer1")
    @Expose
    var pos_voucher_footer1: String? = null

    @ColumnInfo(name = "pos_voucher_footer2")
    @SerializedName("pos_voucher_footer2")
    @Expose
    var pos_voucher_footer2: String? = null

    @ColumnInfo(name = "is_stock_auto_generate")
    @SerializedName("is_stock_auto_generate")
    @Expose
    var stock_auto_generate: String? = null

    @ColumnInfo(name = "pc_count")
    @SerializedName("pc_count")
    @Expose
    var pc_count: String? = null

    @ColumnInfo(name = "expired_month")
    @SerializedName("expired_month")
    @Expose
    var expired_month: String? = null

    @ColumnInfo(name = "paid_status")
    @SerializedName("paid_status")
    @Expose
    var paid_status: String? = null

    @ColumnInfo(name = "h1")
    @SerializedName("h1")
    @Expose
    var h1: String? = null

    @ColumnInfo(name = "h2")
    @SerializedName("h2")
    @Expose
    var h2: String? = null

    @ColumnInfo(name = "h3")
    @SerializedName("h3")
    @Expose
    var h3: String? = null

    @ColumnInfo(name = "h4")
    @SerializedName("h4")
    @Expose
    var h4: String? = null

    @ColumnInfo(name = "f1")
    @SerializedName("f1")
    @Expose
    var f1: String? = null

    @ColumnInfo(name = "f2")
    @SerializedName("f2")
    @Expose
    var f2: String? = null

    @ColumnInfo(name = "f3")
    @SerializedName("f3")
    @Expose
    var f3: String? = null

    @ColumnInfo(name = "f4")
    @SerializedName("f4")
    @Expose
    var f4: String? = null

    @ColumnInfo(name = "tax")
    @SerializedName("tax")
    @Expose
    var tax: String? = null

    @ColumnInfo(name = "branch_code")
    @SerializedName("branch_code")
    @Expose
    var branch_code: String? = null

    @ColumnInfo(name = "branch_name")
    @SerializedName("branch_name")
    @Expose
    var branch_name: String? = null

    @ColumnInfo(name = "hb_code")
    @SerializedName("hb_code")
    @Expose
    var hb_code: String? = null

    @ColumnInfo(name = "credit_sale")
    @SerializedName("credit_sale")
    @Expose
    var credit_sale: String? = null

    @ColumnInfo(name = "use_combo")
    @SerializedName("use_combo")
    @Expose
    var use_combo: String? = null

    @ColumnInfo(name = "last_day_close_date")
    @SerializedName("last_day_close_date")
    @Expose
    var last_day_close_date: String? = null

    @ColumnInfo(name = "last_update_invoice_date")
    @SerializedName("last_update_invoice_date")
    @Expose
    var last_update_invoice_date: String? = null

    @ColumnInfo(name = "company_type")
    @SerializedName("company_type")
    @Expose
    var company_type: String? = null

    @ColumnInfo(name = "company_code")
    @SerializedName("company_code")
    @Expose
    var company_code: String? = null

    @ColumnInfo(name = "start_time")
    @SerializedName("start_time")
    @Expose
    var start_time: String? = null

    @ColumnInfo(name = "end_time")
    @SerializedName("end_time")
    @Expose
    var end_time: String? = null

    @ColumnInfo(name = "branch_id")
    @SerializedName("branch_id")
    @Expose
    var branch_id: String? = null

    @ColumnInfo(name = "print_copy")
    @SerializedName("print_copy")
    @Expose
    var print_copy: String? = null

    @ColumnInfo(name = "tax_type")
    @SerializedName("tax_type")
    @Expose
    var tax_type: String? = null

    @ColumnInfo(name = "balance_control")
    @SerializedName("balance_control")
    @Expose
    var balance_control: String? = null

    @ColumnInfo(name = "transaction_auto_generate")
    @SerializedName("transaction_auto_generate")
    @Expose
    var transaction_auto_generate: String? = null

    @ColumnInfo(name = "company_tax_reg_no")
    @SerializedName("company_tax_reg_no")
    @Expose
    var company_tax_reg_no: String? = null
}